### PR TITLE
with constructor

### PR DESCRIPTION
**Pull request by Srdjanqwe**

### CODE REVIEW

File: src/test/java/pages/SwagLabsLoginPage.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    @FindBy(id ="user-name") WebElement inputUserField;
+    @FindBy(id ="password") WebElement inputPassField;
+    @FindBy(id ="login-button") WebElement loginBtn;
+
+    WebDriver driver;
+    WebDriverWait wdWait;
+
+    public SwagLabsLoginPage(WebDriver driver, WebDriverWait wdWait)
+    {
```
Comment (UkiWoDao) [inline src/test/java/pages/SwagLabsLoginPage.java]: The java convention is to have the opening braces on the same line as the parameter declaration

File: src/test/java/pages/SwagLabsLoginPage.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        this.driver = driver;
+        this.wdWait = wdWait;
+        PageFactory.initElements(driver,this);
+    }
+
+    public void loginAsStandardUser(String username, String password) {
```
Comment (UkiWoDao) [inline src/test/java/pages/SwagLabsLoginPage.java]: why is this method public?

File: src/test/java/pages/SwagLabsLoginPage.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        this.inputUserField.clear();
+        this.inputUserField.sendKeys(username);
+        this.inputPassField.clear();
+        this.inputPassField.sendKeys(password);
+        loginBtn.click();
+    }
```
Comment (UkiWoDao) [inline src/test/java/pages/SwagLabsLoginPage.java]: formatting

File: src/test/java/pages/SwagLabsLoginPage.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    public void fetchCredentialsForStandardUser() {
```
Comment (UkiWoDao) [inline src/test/java/pages/SwagLabsLoginPage.java]: why is this method called "fetchCredentials..."?

File: src/test/java/tests/SwagLabsLoginTest.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import org.junit.jupiter.api.*;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import pages.SwagLabsLoginPage;
+
+import java.time.Duration;
+
+
+public class SwagLabsLoginTest
```
Comment (UkiWoDao) [inline src/test/java/tests/SwagLabsLoginTest.java]: No need for this class to be public. Preferably, use the java conventions for curly brace positions. If you prefer the C#  style, then stick to that convention

**Review by UkiWoDao — CHANGES_REQUESTED** [review]
1. Consider having your page object pages public methods return page objects as well. This would allow you to chain calls and also not pass around the driver in your tests more than once
2. In our particular case, it's practically certain that the login would execute very fast. In live projects, the page hosting the login mechanism would be a good candidate for being extended with LoadableComponent<> class, since it would tend to take some time, and get slower over time

**Review by shelfsrdjan — APPROVED** [review]

**Pull request merged**